### PR TITLE
Slave instance UI updates

### PR DESCRIPTION
**Pull request by bbassie**
Because the other branch broke on my end, here is a new one 🤦‍♂️ 

https://github.com/clusterio/clusterio/pull/423

### CODE REVIEW

**Comment by Danielv123** [comment]
Hi, any progress on fixing the issues mentioned by Hornwitser?

**Comment by bbassie** [comment]
They should be fixed in this one

File: packages/web_ui/src/components/SlaveViewPage.jsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import React, { useContext, useState } from "react";
-import { Descriptions, PageHeader, Spin, Typography } from "antd";
+import { Descriptions, PageHeader, Spin, Tag, Typography } from "antd";
```
Comment (Danielv123) [inline packages/web_ui/src/components/SlaveViewPage.jsx:2]: Haha, love how you sorted the imports alphabetically between PRs XD

File: packages/web_ui/src/components/InstanceStatusTag.jsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { Tag } from "antd";
+import React from "react";
+
+const statusColors = {
+	unknown: "#d46b08",
+	unassigned: "#d4380d",
+	stopped: "#cf1322",
+	starting: "#7cb305",
```
Comment (Danielv123) [inline packages/web_ui/src/components/InstanceStatusTag.jsx:8]: After running this on my dev cluster for a few days I have found the colors for starting and running to be a bit hard to tell appart. colors

**Review by Danielv123 — APPROVED** [review]
Looks like you got all the changes requested by Hornwitser. I have been running this on my development environment for a while now, and it certainly is nice to have.

File: packages/web_ui/src/components/InstanceStatusTag.jsx (hunk 2 above)
Comment (psihius) [reply packages/web_ui/src/components/InstanceStatusTag.jsx:8]: Maybe skew the "starting" closer to yellow?

**Review by psihius — APPROVED** [review]

File: packages/web_ui/src/components/InstanceStatusTag.jsx (hunk 2 above)
Comment (bbassie) [reply packages/web_ui/src/components/InstanceStatusTag.jsx]: Changed the colors to these:
image

File: packages/web_ui/src/components/InstanceStatusTag.jsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { Tag } from "antd";
+import React from "react";
+
+const statusColors = {
+	unknown: "#d46b08",
+	unassigned: "#d4380d",
+	stopped: "#cf1322",
+	starting: "#d4b106",
+	running: "#52c41a",
+	stopping: "#d48806",
```
Comment (psihius) [inline packages/web_ui/src/components/InstanceStatusTag.jsx:10]: How about something like this for stopping - an orange color: #ff9f29

Comment (psihius) [reply packages/web_ui/src/components/InstanceStatusTag.jsx:10]: For the unknown may be some other colour would fit, something that stands out and not used other for statuses. For example: #ff00f7
And for unnasigned, maybe something gray?

**Comment by bbassie** [comment]
Updated not found:
image

Only visible when you view an instance page and another user deletes it:
image

**Comment by bbassie** [comment]
@Danielv123 @Hornwitser All should be ready now 🤞

**Review by psihius — APPROVED** [review]

**Review by Danielv123 — APPROVED** [review]

**Comment by Danielv123** [comment]
Merged the latest changes into my dev branch. Didn't have any issues running master on dev branch with slave on master branch. New colors are a big improvement.

File: packages/slave/slave.js [hunk 4]
```diff
@@ -260,7 +260,9 @@ class Instance extends libLink.Link {
 	notifyStatus(status) {
 		this._status = status;
 		libLink.messages.instanceStatusChanged.send(this, {
-			instance_id: this.id, status,
+			instance_id: this.id,
+			status,
+			game_port: this.server?.gamePort || this.config.get("factorio.game_port") || null,
```
Comment (Hornwitser) [inline packages/slave/slave.js:265]: The optional chaining operator is [not supported in Node v12](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#browser_compatibility) and may not be used.

**Pull request merged**